# Config format schema number
format_version: 4

###################
## Model options
model_params:
  model_architecture: "cylinder_asym"

  output_shape: #体素大小
    - 480
    - 360
    - 32

  fea_dim: 9
  out_fea_dim: 256
  num_class: 20
  num_input_features: 16 #16 width (more channels) 
  use_norm: True
  init_size: 8 #16 #超参！


###################
## Dataset options
dataset_params:
  dataset_type: "cylinder_dataset"
  pc_dataset_type: "SemKITTI_sk"
  ignore_label: 0
  return_test: False # False
  fixed_volume_space: True
  label_mapping: "config/label_mapping/semantic-kitti.yaml"
  max_volume_space:
    - 50
    - 3.1415926
    - 2
  min_volume_space:
    - 0
    - -3.1415926
    - -4


###################
## Data_loader options
train_data_loader:
  data_path: "/media/oliver/Elements SE/KITTI/sequences"
  imageset: "train"
  return_ref: True
  batch_size: 3 #2
  shuffle: True
  num_workers: 12 #4

val_data_loader:
  data_path: "/media/oliver/Elements SE/KITTI/sequences"
  imageset: "val" #"val" #"test"
  return_ref: True
  batch_size: 1
  shuffle: False
  num_workers: 12 #4


###################
## Train params
train_params:
  model_load_path: "model_load_dir/model_full_ft8.pt"
  model_save_path: "model_save_dir/model_tmp.pt"
  checkpoint_every_n_steps: 4599
  max_num_epochs: 20 #40
  eval_every_n_steps: 5000 #4599
  learning_rate: 0.002 #1
  weight_decay: 0.001
  mixed_fp16: True
